rfc/rfc-36/rfc-36.md

```diff
@@ -0,0 +1,605 @@
+      | is_partition_col | tinyint | whether the column is a partition column |
+      | comment          | string  | column description                       |
+
+##### Partition
+
+- **partitions**
+
+    -  unique key: part_name, tbl_id
+
+    - | name        | type      | comment                          |
+          | ----------- | --------- | -------------------------------- |
+      | part_id     | bigint    | auto_increment                   |
+      | tbl_id      | bigint    | table the partition belongs to   |
+      | part_name   | string    | partition name                   |
+      | create_time | timestamp | partition created time           |
+      | update_time | timestamp | partition updated time           |
+      | is_deleted  | tinyint   | whether the partition is deleted |
+
+- **partition****_parmas**
+
+    -  unique key: param_key, part_id
+
+    - | name        | type   | comment                        |
+          | ----------- | ------ | ------------------------------ |
+      | part_id     | bigint | partition identifier           |
+      | tbl_id      | bigint | table the partition belongs to |
+      | param_key   | string | partition parameter key        |
+      | param_value | string | partition parameter value      |
+
+- **partition****_key_val**
+
+    -  is used to support partition pruning.
+
+    -  unique key: part_key_name, part_id
+
+    - | name          | type   | comment                        |
+          | ------------- | ------ | ------------------------------ |
+      | part_id       | bigint | partition  identifier          |
+      | tbl_id        | bigint | table the partition belongs to |
+      | part_key_name | string | the partition name             |
+      | part_key_val  | string | the partition value            |
+
+##### Timeline
+
+- **tbl_timestamp**
+
+    -  is used to get a self-increasing timestamp for each table
+
+    -  unique key: part_key_name, part_id
+
+    - | name   | type   | comment               |
+          | ------ | ------ | --------------------- |
+      | tbl_id | bigint | table identifier      |
+      | ts     | string | timestamp for instant |
+
+- **instant**
+
+    -  unique key: ts, action, tbl_id
+
+    - | name       | type      | comment                                      |
+          | ---------- | --------- | -------------------------------------------- |
+      | instant_id | bigint    | auto_increment                               |
+      | tbl_id     | bigint    | table identifier                             |
+      | ts         | string    | instant timestamp                            |
+      | action     | tinyint   | instant action, commit, deltacommit and etc. |
+      | state      | tinyint   | instant state, requested, inflight and etc.  |
+      | duration   | int       | for heartbeat                                |
+      | start_ts   | timestamp | for heartbeat                                |
```

Comment:
What is the reason behind having 2 different fields for heartbeat? Can you please elaborate?

Reply:
Whether the heartbeat fails, it is judeged by whether duration + start_ts < current_time.